sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/FieldBuilder.java

```diff
@@ -82,31 +87,48 @@
         classChain.push(currentClass);
         List<SearchField> searchFields = Arrays.stream(currentClass.getDeclaredFields())
             .filter(classField -> !classField.isAnnotationPresent(FieldIgnore.class))
-            .map(classField -> buildField(classField, classChain, logger))
+            .map(classField -> buildField(classField, classChain, serializer, logger))
+            .filter(Objects::nonNull)
             .collect(Collectors.toList());
         classChain.pop();
         return searchFields;
     }
 
-    private static SearchField buildField(java.lang.reflect.Field classField, Stack<Class<?>> classChain,
-        ClientLogger logger) {
+    private static SearchField buildField(Field classField, Stack<Class<?>> classChain,
+        PropertyNameSerializer serializer, ClientLogger logger) {
+        if (serializer == null) {
+            serializer = new JacksonJsonSerializerProvider().createInstance();
+        }
```

Comment:
We should create this outside of the stream, right now this is going to construct a new instance for each declared field.